web/src/main/java/com/graphhopper/http/cli/ExportCommand.java

```diff
@@ -144,16 +145,22 @@ private static void writeStreetEdgesCsv(GraphHopper configuredGraphHopper,
             // Convert GH's distance in meters to millimeters to match R5's implementation
             long distanceMillimeters = distanceMeters * 1000;
 
-            // Fetch OSM ID and accessibility flags for each edge direction
+            // Fetch OSM ID, skipping edges from PT meta-graph that have no IDs set (getOsmIdForGhEdge returns -1)
```

Comment:
Just a heads up - this "skipped edge" code has been in the mega-branch (ie, it's been used in all of our test runs so far) for quite a while; I guess I hadn't realized that it hadn't made it into master until now. After this gets merged, I'll try to merge all changes still left over in the mega-branch so we can get rid of that for good and work off of good old `original-direction`

Reply:
Interesting... what does it do exactly? what does it mean for an edge to have no id? What would we expect to see in the output?

Reply:
These are essentially "fake"/"hidden" edges that GraphHopper needs to add to the graph in order to do PT routing, but they never get returned by the GH servers - they're only used by GH internally for dealing with PT access/egress from an algorithmic standpoint. See the figure [here](https://github.com/graphhopper/graphhopper/blob/master/reader-gtfs/pt-model.png) for more of an idea of what this network looks like.

Because this network isn't part of the actual street network, and we never deal with these edges directly when making PT routing requests, we don't want them in the exported CSV file (it turns out that there are many more of these PT-only edges than normal street edges based on how this "meta-graph" needs to be constructed, so including them in the CSV results in a size blowup of >>2x).

Reply:
(and you can see that I've "marked" these edges - edges that didn't have an OSM ID associated with them, ie, these transit-specific edges -  purposefully with an id of `-1` [here](https://github.com/replicahq/graphhopper/pull/32/files#diff-2af315a4a867d9346046b1c6fcf2e0adR54). That's why this check in particular is looking to see if the returned OSM id `== -1L`)